### DISCUSSION TITLE
Keeping the Original File Attributes when decrypting.

### DISCUSSION THREAD

**Discussion by OA971** [FiloSottile/age#468] [General]
I don't know if this is easy or difficult to be done, but if you can keep the original file attributes to be decrypted with it that would be perfect.
For Example: Date & time of the Original File.

**Comment by str4d** [comment]
Preserving metadata would require the age file format to have metadata fields for storing information about the file. It does not have this capability, for two reasons:

- If the metadata were stored in plaintext as part of the age file format, then that would leak information about the encrypted file.
- If the metadata were stored encrypted (as we would want) as part of the age file format, then it is equivalent to just encrypting a "wrapping file" that houses the original file and its metadata together.

And there is already a suitable "wrapping file" type: `tar`! We can create a `tar`archive containing a single file, and encrypt that:

```shell
tar c MY_FILE | age -r RECIPIENT -o MY_FILE.tar.age
```
